mlir/lib/Dialect/MemRef/IR/MemRefOps.cpp

```diff
@@ -2223,10 +2223,6 @@ LogicalResult ExpandShapeOp::verify() {
   MemRefType srcType = getSrcType();
   MemRefType resultType = getResultType();
 
-  if (srcType.getRank() >= resultType.getRank())
```

Comment:
We still have to fail verification if `srcType.getRank() > resultType.getRank()`.

Reply:
It's not really possible to do this, because of the reassociation indices. I've added some tests to illustrate this in the (upcoming) update.